reframe/core/config.py

```diff
@@ -625,7 +625,7 @@ def _find_config_file():
     return None
 
 
-def load_config(filename=None):
+def load_config(filename=None, autodetect_opts=None):
```

Comment:
Instead of passing a dictionary of options here, I would rather add an argument about the method and then the rest of keyword arguments would be the autodetection options.

```suggestion
def load_config(filename=None, autodetect_method='hostname', **autodetect_opts):
```